firmware/shared/src/hw/hw_sdio_sd.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "hw_hal.h"
+#include <stdbool.h>
+#include <stdint.h>
+
+// Check here for documentation about the type defined by HAL
+// https://www.disca.upv.es/aperles/arm_cortex_m3/llibre/st/STM32F439xx_User_Manual/structsd__handletypedef.html
+
+typedef struct
+{
+    SD_HandleTypeDef *const hsd;     // the HAL SD handle that will hold the state of the SD card
+    uint32_t                timeout; // the timeout for the SD card operations
+} SD_card;                           // struct that included all the state about SDIO and SD card
+
+typedef struct
+{
+    SD_TypeDef *   sdio;        // sdio register base address
+    SD_InitTypeDef sdio_config; // sdio config
+    uint32_t       timeout;     // timeout config
+} SD_card_init_config;
+
+typedef enum
+{
+    SD_card_OK      = HAL_OK,
+    SD_card_ERROR   = HAL_ERROR,
+    SD_card_BUSY    = HAL_BUSY,
+    SD_card_TIMEOUT = HAL_TIMEOUT
+} SD_card_status; // wrapper of HAL status for better interface
```

Comment:
Nice, I like that you've wrapped this.
We do `SCREAMING_SNAKE_CASE` for enums, so `SD_CARD_OK`, `SD_CARD_ERROR`, etc.